File: Codigos/C++/C2/Listas e Trabalhos/POLIGONO/POLIGONO/circulo.cpp
#include "circulo.h"
#include "math.h"

Circulo::Circulo(float raio){
    this->raio = raio;
    this->area = 0;
}

void Circulo::Area(){
    this->area = (PI*(pow(this->raio,2)));
}

float Circulo::getArea(){
    return this->area;
}

Circulo::~Circulo(){}

File: Codigos/Codigos/JAVA/WorkSpace C2/FormasGeometricas/src/model/Circulo.java
package model;

public class Circulo extends Poligono {
	
	private Ponto  p;
	private double r;
	
	public Circulo ( int x, int y, double r ) {
		p = new Ponto ( x, y );
		this.r = r;		
	}

	public Circulo ( Ponto p, double r ) {
		this.p = p.clone();
		this.r = r;		
	}		

	@Override
	public double getArea() {
		return Math.PI * Math.pow(r, 2);		
	}
	
	public String toString () {
		return this.p + ", " + r;
	}
	

}
